managed/services/grafana/auth_server.go

```diff
@@ -356,9 +356,14 @@ func (s *AuthServer) getFiltersForVMProxy(userID int) ([]string, error) {
 
 	filters := make([]string, 0, len(roles))
 	for _, r := range roles {
-		if r.Filter != "" {
-			filters = append(filters, r.Filter)
+		if r.Filter == "" {
+			// Special case when a user has assigned a role with no filters.
+			// In this case it's irrelevant what other roles the user has assigned.
+			// The user shall have full access.
+			return nil, nil
```

Comment:
```suggestion
			return nil, nil //nolint:nilnil
```
We have a rule now :)

Reply:
I changed it to return empty slice.
I'm wondering why nothing complained about the `nilnil` rule.

Reply:
Because you haven't rebased after I merged that rule, no?